agent/session.go

```diff
@@ -54,6 +56,7 @@ func newSession(ctx context.Context, agent *Agent, delay time.Duration, sessionI
 		subscriptions: make(chan *api.SubscriptionMessage),
 		registered:    make(chan struct{}),
 		closed:        make(chan struct{}),
+		cancel:        sessionCancel,
```

Comment:
I went with the same session+cancellation because it seems like `newSession` is the only thing that calls `session.run`, which is contingent upon `session.start` succeeding.  Again though, happy to go either way.